internal/server/auth/middleware.go

```diff
@@ -176,6 +177,52 @@ func EmailMatchingInterceptor(logger *zap.Logger, rgxs []*regexp.Regexp) grpc.Un
 	}
 }
 
+func NamespaceMatchingInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		auth := GetAuthenticationFrom(ctx)
+
+		if auth == nil {
+			return handler(ctx, req)
+		}
+
+		// this mechanism only applies to static toke authentications
+		if auth.Method != authrpc.Method_METHOD_TOKEN {
+			return handler(ctx, req)
+		}
```

Comment:
Just thinking out loud (this is probably is good for this pass):

In the future it would be nice to be able to get an associated policy from the authentication.
Instead of the enforcing middleware having to know about the types of authentications, it should just ask for the policy / authorization. Then inspect that common policy structure / interface as to whether or not the request is valid.

That might be a bit over abstracted for this first pass, but will make extending policy to other authentication types easier.